server/src/main/java/org/elasticsearch/index/mapper/XContentDataHelper.java

```diff
@@ -103,19 +117,22 @@ static void decodeAndWrite(XContentBuilder b, BytesRef r) throws IOException {
      * @throws IOException
      */
     static void writeMerged(XContentBuilder b, String fieldName, List<BytesRef> encodedParts) throws IOException {
-        if (encodedParts.isEmpty()) {
+        var partsWithData = encodedParts.stream().filter(XContentDataHelper::isDataPresent).toList();
```

Comment:
iirc stream ops are fairly slow, consider replacing with regular list processing.

Reply:
You can probably be optimistic and first check if there's an empty element, then create the copy on a second pass if one is found.

Reply:
I rewrote this but i don't have a feel if this is any faster. It skips allocating stream object i guess.